distribution.go

```diff
@@ -120,6 +120,18 @@ func distributedFetch[V, T any](c *Client[T], key string, fetchFn FetchFn[V]) Fe
 			c.reportDistributedCacheHit(false)
 		}
 
+		// Before we call the fetchFn, we'll do an unblocking read to see if the
+		// context has been cancelled. If it has, we'll return a stale value if we
+		// have one available.
+		select {
```

Comment:
One could argue that it’s simpler to just perform an error check here, e.g. `if ctx.Err() != nil`, but I just find this approach more explicit. To me it's very clear that this code is trying to determine whether the work should proceed or not.